### PR TITLE
Rails Blog App - Validations and model specs

### PR DESCRIPTION
**Pull request by mailsg**
Hi,

This PR is raised to showcase the implementation of Validations and model specs. Below are the pointers covered in the implementation:

- Added the following validations:
  - For the `User model`:
    - Name must not be blank.
    - `PostsCounter` must be an integer greater than or equal to zero.
  
  - For the `Post model`:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - `CommentsCounter` must be an integer greater than or equal to zero.
    - `LikesCounter` must be an integer greater than or equal to zero.

- Added unit specs for all the models' methods and validations.

Kindly review the PR and provide feedback on the same.

Thanks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -115,7 +115,7 @@ To run the project, execute the following command:
 To run tests, run the following command:
 
 ```sh
-  bin/rails test test/models/article_test.rb
+  rspec
```
Comment (binhussen) [inline README.md:118]: - Kindly, your `mit` link is broken please update it to have a professional readme file 👍

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @mailsg , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Added validations and model specs✔️
- Well Structured Code ✔️
- Used Git flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (mailsg) [reply README.md:118]: 👀

**Review by binhussen — APPROVED** [review]
Hi @mailsg , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉
IApproveSimonCowellGIF (2)

## Highlights
- Added validations and model specs✔️
- Well Structured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**